### PR TITLE
Fix dhcpv6 counter test failure

### PR DESCRIPTION
**Pull request by jcaiMR**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [x] 202205

#### What is the motivation for this PR?
Fix test dhcpv6 counter failure
Dual tor issue:
1. Solicit counter not increase, random select dut will select standby port dut
2. Advertise counter not increase, downlink mac is not right, should always dut_mac

Single Tor issue:
Decline/Release/Information request/unknow counter not increase
test case send 14 kinds of packets almost at the same time.
As it's multicast packets, each packet get duplicated on fanout switch (maybe due to some topo setting), on one of local DUT,
observed 23 times duplicated packets (around 270 pakcets) send in 22 ms.
By default raw socket recv buffer size is 32k (270x128B = 34.5K), so some kinds of packets (always the packets at the last sequence like Release, Decline, Information request ...) get drooped and cause test case failure.   

image

image

#### How did you do it?
1) toggle_all_simulator_ports_to_upper_tor and always use upper tor to avoid mux status inconsistent issue 
2) use dut_mac for downlink interface mac (always same as port channel mac)
3) Add 1 second delay during testing for each kind of packets, in case any multicast packet duplicate issue happened 

#### How did you verify/test it?
Test on single tor and dual tor DUT.

### CODE REVIEW

File: ansible/roles/test/files/ptftests/py3/dhcpv6_counter_test.py [hunk 1]
```diff
@@ -137,10 +144,12 @@ def create_unknown_server_packet(self):
     """
 
     def client_send(self):
-        client_messages = [DHCP6_Solicit, DHCP6_Request, DHCP6_Confirm, DHCP6_Renew, DHCP6_Rebind, DHCP6_Release, DHCP6_Decline, DHCP6_Reconf, DHCP6_InfoRequest]
+        client_messages = [DHCP6_Solicit, DHCP6_Request, DHCP6_Confirm, DHCP6_Renew,
+                           DHCP6_Rebind, DHCP6_Release, DHCP6_Decline, DHCP6_InfoRequest]
         for message in client_messages:
             packet = self.create_packet(message)
             testutils.send_packet(self, self.client_port_index, packet)
+            time.sleep(1)
```
Comment (kellyyeh) [inline ansible/roles/test/files/ptftests/py3/dhcpv6_counter_test.py:152]: Can you add a comment here for why one second is needed?

File: ansible/roles/test/files/ptftests/py3/dhcpv6_counter_test.py [hunk 2]
```diff
@@ -92,41 +100,40 @@ def generate_client_interace_ipv6_link_local_address(self, client_port_index):
...
 
-
     """
      Packet generation functions/wrappers
 
     """
 
     def create_packet(self, message):
-        packet = Ether(src=self.client_mac, dst=self.BROADCAST_MAC)
+        packet = ptf.packet.Ether(src=self.client_mac, dst=self.BROADCAST_MAC)
```
[the remark this answers is not in the document]
Comment (jcaiMR) [reply ansible/roles/test/files/ptftests/py3/dhcpv6_counter_test.py:109]: In order to pass python check, python check need give the clear lib name.

File: ansible/roles/test/files/ptftests/py3/dhcpv6_counter_test.py (hunk 1 above)
Comment (jcaiMR) [reply ansible/roles/test/files/ptftests/py3/dhcpv6_counter_test.py:152]: In order to avoid single tor issue I mentioned in description. Sometime multicast packets flooded to all ports and make filter socket RECV socket full and finally cause packet drop. So add 1 second here is a work around for this kind of cases (to low down the packet send rate)

Comment (jcaiMR) [reply ansible/roles/test/files/ptftests/py3/dhcpv6_counter_test.py:155]: Applied in latest diff.

**Review by kellyyeh — DISMISSED** [review]

File: tests/dhcp_relay/test_dhcpv6_relay.py [hunk 3]
```diff
@@ -24,6 +23,14 @@
 
 logger = logging.getLogger(__name__)
 
+
+def wait_all_bgp_up(duthost):
+    config_facts = duthost.config_facts(host=duthost.hostname, source="running")['ansible_facts']
+    bgp_neighbors = config_facts.get('BGP_NEIGHBOR', {})
+    if not wait_until(300, 10, 0, duthost.check_bgp_session_state, bgp_neighbors.keys()):
```
Comment (StormLiangMS) [inline tests/dhcp_relay/test_dhcpv6_relay.py:30]: is that 300 seconds delay too long? maybe 60 seconds do the job? @jcaiMR

**Review by StormLiangMS — APPROVED** [review]
Approved with a minor suggestion.

**Review by StormLiangMS — APPROVED** [review]
LGTM

**Review by kellyyeh — APPROVED** [review]

**Comment by wangxin** [comment]
@jcaiMR Cherry-picking this PR to 202012 branch has conflicts that are hard to resolve. Can you create separate PR to 202012 branch to have this change?
